projects/plugins/jetpack/_inc/blogging-prompts.php

```diff
@@ -18,3 +18,180 @@ function jetpack_blogging_prompts_add_meta_data( $keys ) {
 	$keys[] = '_jetpack_blogging_prompt_key';
 	return $keys;
 }
+
+/**
+ * Sets up a new post as an answer to a blogging prompt.
+ *
+ * Called on `wp_insert_post` hook.
+ *
+ * @param int $post_id ID of post being inserted.
+ * @return void
+ */
+function jetpack_setup_blogging_prompt_response( $post_id ) {
+	$prompt_id = isset( $_GET['answer_prompt'] ) && absint( $_GET['answer_prompt'] ) ? absint( $_GET['answer_prompt'] ) : false; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+	if ( ! jetpack_is_new_post_screen() || ! $prompt_id ) {
+		return;
+	}
+
+	if ( jetpack_is_valid_blogging_prompt( $prompt_id ) ) {
+		update_post_meta( $post_id, '_jetpack_blogging_prompt_key', $prompt_id );
+		wp_add_post_tags( $post_id, 'dailyprompt' );
+	}
+}
+
+add_action( 'wp_insert_post', 'jetpack_setup_blogging_prompt_response' );
+
+/**
+ * Determines if the current screen (in wp-admin) is creating a new post.
+ *
+ * /wp-admin/post-new.php
+ *
+ * @return bool
+ */
+function jetpack_is_new_post_screen() {
+	global $current_screen;
+
+	if (
+		$current_screen instanceof \WP_Screen &&
+		'add' === $current_screen->action &&
+		'post' === $current_screen->post_type
+	) {
+		return true;
+	}
+
+	return false;
+}
+
+/**
+ * Checks if the given prompt id is included in today's blogging prompts.
+ *
+ * Would be best to use the API to check if the prompt id is valid for any day,
+ * but for now we're only using one prompt per day.
+ *
+ * @param int $prompt_id id of blogging prompt.
+ * @return bool
+ */
+function jetpack_is_valid_blogging_prompt( $prompt_id ) {
+	$daily_prompts = jetpack_get_daily_blogging_prompts();
+
+	if ( ! $daily_prompts ) {
+		return false;
+	}
+
+	foreach ( $daily_prompts as $prompt ) {
+		if ( $prompt->id === $prompt_id ) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+/**
+ * Determines if the site might have a blog.
+ *
+ * @param @param array $checks {
+ *     Optional. An array of checks to determine if the site has a blog.
+ *
+ *     @type bool $post_count  Whether to check the number of posts. Default true.
+ *     @type bool $posts_page  Whehter to check if posts page is set. Default true.
+ *     @type bool $site_intent Whether to check the site intent when the site was created. Default true.
+ * }
+ * @return bool
+ */
+function jetpack_is_potential_blogging_site( $checks = array() ) {
```

Comment:
It sounds like we can just reuse the same logic asw the mobile app p1667829421316519/1667217622.476049-slack-C03NLNTPZ2T